src/vendor.rs

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    pub fn call4(
+        eid: u32,
+        fid: u32,
+        arg0: u64,
+        arg1: u64,
+        arg2: u64,
+        arg3: u64,
+        arg4: u64,
+    ) -> Self {
+        VendorFunction::Vendor {
+            eid: (eid),
+            fid: (fid),
+            arg0: (arg0),
+            arg1: (arg1),
+            arg2: (arg2),
+            arg3: (arg3),
+            arg4: (arg4),
+            arg5: (0),
+        }
+    }
+
+    pub fn call5(
+        eid: u32,
+        fid: u32,
+        arg0: u64,
+        arg1: u64,
+        arg2: u64,
+        arg3: u64,
+        arg4: u64,
+        arg5: u64,
+    ) -> Self {
+        VendorFunction::Vendor {
+            eid: (eid),
+            fid: (fid),
+            arg0: (arg0),
+            arg1: (arg1),
+            arg2: (arg2),
+            arg3: (arg3),
+            arg4: (arg4),
+            arg5: (arg5),
+        }
+    }
+}
+
+impl SbiFunction for VendorFunction {
+    fn a0(&self) -> u64 {
+        match self {
+            VendorFunction::Vendor {
+                eid: _,
+                fid: _,
+                arg0,
+                arg1: _,
+                arg2: _,
+                arg3: _,
+                arg4: _,
+                arg5: _,
+            } => *arg0 as u64,
+        }
+    }
+
+    fn a1(&self) -> u64 {
+        match self {
+            VendorFunction::Vendor {
+                eid: _,
+                fid: _,
+                arg0: _,
+                arg1,
+                arg2: _,
+                arg3: _,
+                arg4: _,
+                arg5: _,
+            } => *arg1 as u64,
+        }
+    }
+
+    fn a2(&self) -> u64 {
+        match self {
+            VendorFunction::Vendor {
+                eid: _,
+                fid: _,
+                arg0: _,
+                arg1: _,
+                arg2,
+                arg3: _,
+                arg4: _,
+                arg5: _,
+            } => *arg2 as u64,
+        }
+    }
+
+    fn a3(&self) -> u64 {
+        match self {
+            VendorFunction::Vendor {
+                eid: _,
+                fid: _,
+                arg0: _,
+                arg1: _,
+                arg2: _,
+                arg3,
+                arg4: _,
+                arg5: _,
+            } => *arg3 as u64,
+        }
+    }
+    fn a4(&self) -> u64 {
+        match self {
+            VendorFunction::Vendor {
+                eid: _,
+                fid: _,
+                arg0: _,
+                arg1: _,
+                arg2: _,
+                arg3: _,
+                arg4,
+                arg5: _,
+            } => *arg4 as u64,
+        }
+    }
+
+    fn a5(&self) -> u64 {
+        match self {
+            VendorFunction::Vendor {
+                eid: _,
+                fid: _,
+                arg0: _,
+                arg1: _,
+                arg2: _,
+                arg3: _,
+                arg4: _,
+                arg5,
+            } => *arg5 as u64,
```

Comment:
These are incredibly verbose - and my first thought would be to turn these into a macro using paste! or similar. I'd appreciate comments.